Firmware/Marlin_main.cpp

```diff
@@ -9573,7 +9573,7 @@ void prepare_move(uint16_t start_segment_idx)
 void prepare_arc_move(bool isclockwise, uint16_t start_segment_idx) {
     float r = hypot(offset[X_AXIS], offset[Y_AXIS]); // Compute arc radius for mc_arc
     // Trace the arc
-    mc_arc(current_position, destination, offset, feedrate * feedmultiply / 60 / 100.0, r, isclockwise, start_segment_idx);
+    mc_arc(current_position, destination, offset, (feedrate * feedmultiply) / 6000.f, r, isclockwise, start_segment_idx);
```

Comment:
actually this refactored implementation look more correct to me than the previous one, because:
- `feedrate` is a `float`
- `feedmultiply` is an `int`
- `60` was an `int`
- `100.0` was a 'double`

`feedrate * feedmultiply` will surely not overflow in our application and dividing it by `6000.f` will not clamp it to `int` anywhere on the way, so :+1:

Reply:
Theoretically, we might get a little faster by getting rid of the division as well:
```
    mc_arc(current_position, destination, offset, (feedrate * feedmultiply) * (1.f/6000.f), r, isclockwise, start_segment_idx);
		lds r22, <feedmultiply>
		lds r23, <feedmultiply+0x1>
		mov r0, r23   
		add r0, r0   
		sbc r24, r24   
		sbc r25, r25   
		call <__floatsisf>
		lds r18, <feedrate>
		lds r19, <feedrate+0x1>
		lds r20, <feedrate+0x2>
		lds r21, <feedrate+0x3>
		call <__mulsf3>
		ldi r18, 0x3E ; 62 
		ldi r19, 0xC3 ; 195 
		ldi r20, 0x2E ; 46 
		ldi r21, 0x39 ; 57 
		call <__mulsf3>
		adiw r28, 0x2a ; 42 
		std Y+60, r22 ; 0x3c 
		std Y+61, r23 ; 0x3d 
		std Y+62, r24 ; 0x3e 
		std Y+63, r25 ; 0x3f 
		sbiw r28, 0x2a ; 42 
```

Reply:
That's a good idea. I thought the compiler didn't make something like this a compile time constant? Interesting.

Reply:
Nice I can confirm this on my end, I will submit another commit with this change :)

Reply:
I'm surprised the compiler refuses to do such optimizations. May be we don't have the speculative math optimizations turned on (and I bet for a good reason).